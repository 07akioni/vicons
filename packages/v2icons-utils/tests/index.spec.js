const { Accessibility16Regular } = require('../../xicons/dist/fluent/vue2/lib')
const { AccessibleIcon } = require('../../xicons/dist/fa/vue2/lib')
const { AcUnitFilled } = require('../../xicons/dist/material/vue2/lib')
const { IosAdd } = require('../../xicons/dist/ionicons4/vue2/lib')
const { Accessibility } = require('../../xicons/dist/ionicons5/vue2/lib')
const { AB } = require('../../xicons/dist/tabler/vue2/lib')
const { AccountBookFilled } = require('../../xicons/dist/antd/vue2/lib')
const { mount } = require('@vue/test-utils')

;[
  ['fluent', Accessibility16Regular],
  ['fa', AccessibleIcon],
  ['material', AcUnitFilled],
  ['ionicons4', IosAdd],
  ['ionicons5', Accessibility],
  ['tabler', AB],
  ['antd', AccountBookFilled],
].forEach(([ name, icon ]) => {
  describe(name,() => {
    it('works', () => {
      expect(mount(icon).html()).toMatchSnapshot()
    })
  })
})
