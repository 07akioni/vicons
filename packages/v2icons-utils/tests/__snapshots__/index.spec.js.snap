// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`antd works 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" viewBox=\\"0 0 1024 1024\\">
  <path d=\\"M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zM648.3 426.8l-87.7 161.1h45.7c5.5 0 10 4.5 10 10v21.3c0 5.5-4.5 10-10 10h-63.4v29.7h63.4c5.5 0 10 4.5 10 10v21.3c0 5.5-4.5 10-10 10h-63.4V752c0 5.5-4.5 10-10 10h-41.3c-5.5 0-10-4.5-10-10v-51.8h-63.1c-5.5 0-10-4.5-10-10v-21.3c0-5.5 4.5-10 10-10h63.1v-29.7h-63.1c-5.5 0-10-4.5-10-10v-21.3c0-5.5 4.5-10 10-10h45.2l-88-161.1c-2.6-4.8-.9-10.9 4-13.6c1.5-.8 3.1-1.2 4.8-1.2h46c3.8 0 7.2 2.1 8.9 5.5l72.9 144.3l73.2-144.3a10 10 0 0 1 8.9-5.5h45c5.5 0 10 4.5 10 10c.1 1.7-.3 3.3-1.1 4.8z\\" fill=\\"currentColor\\"></path>
</svg>"
`;

exports[`fa works 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" viewBox=\\"0 0 448 512\\">
  <path d=\\"M423.9 255.8L411 413.1c-3.3 40.7-63.9 35.1-60.6-4.9l10-122.5l-41.1 2.3c10.1 20.7 15.8 43.9 15.8 68.5c0 41.2-16.1 78.7-42.3 106.5l-39.3-39.3c57.9-63.7 13.1-167.2-74-167.2c-25.9 0-49.5 9.9-67.2 26L73 243.2c22-20.7 50.1-35.1 81.4-40.2l75.3-85.7l-42.6-24.8l-51.6 46c-30 26.8-70.6-18.5-40.5-45.4l68-60.7c9.8-8.8 24.1-10.2 35.5-3.6c0 0 139.3 80.9 139.5 81.1c16.2 10.1 20.7 36 6.1 52.6L285.7 229l106.1-5.9c18.5-1.1 33.6 14.4 32.1 32.7zm-64.9-154c28.1 0 50.9-22.8 50.9-50.9C409.9 22.8 387.1 0 359 0c-28.1 0-50.9 22.8-50.9 50.9c0 28.1 22.8 50.9 50.9 50.9zM179.6 456.5c-80.6 0-127.4-90.6-82.7-156.1l-39.7-39.7C36.4 287 24 320.3 24 356.4c0 130.7 150.7 201.4 251.4 122.5l-39.7-39.7c-16 10.9-35.3 17.3-56.1 17.3z\\" fill=\\"currentColor\\"></path>
</svg>"
`;

exports[`fluent works 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" viewBox=\\"0 0 16 16\\">
  <g fill=\\"none\\">
    <path d=\\"M8 1a2.25 2.25 0 0 0-2.184 2.794L3.977 3.1A1.605 1.605 0 0 0 2.79 6.08L5 7.005v2.391l-1.159 3.48a1.606 1.606 0 0 0 3.03 1.065l1.121-3.027l1.125 3.024a1.605 1.605 0 0 0 3.029-1.062L11 9.397V7.001l2.184-.906a1.612 1.612 0 0 0-1.19-2.996l-1.808.686A2.254 2.254 0 0 0 8 1zM6.75 3.25a1.25 1.25 0 1 1 2.5 0a1.25 1.25 0 0 1-2.5 0zm5.6.784a.612.612 0 0 1 .45 1.138l-2.492 1.033a.5.5 0 0 0-.308.462v2.81a.5.5 0 0 0 .025.157l1.171 3.555a.605.605 0 0 1-1.142.4l-1.19-3.202c-.301-.808-1.444-.807-1.743.001l-1.188 3.205a.606.606 0 0 1-1.143-.402l1.184-3.556A.5.5 0 0 0 6 9.477V6.672a.5.5 0 0 0-.307-.462L3.176 5.158a.605.605 0 0 1 .447-1.124l3.572 1.351c.513.195 1.08.194 1.593 0l3.562-1.35z\\" fill=\\"currentColor\\"></path>
  </g>
</svg>"
`;

exports[`ionicons4 works 1`] = `
"<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 512 512\\" enable-background=\\"new 0 0 512 512\\" xml:space=\\"preserve\\">
  <g>
    <path d=\\"M368.5,240H272v-96.5c0-8.8-7.2-16-16-16c-8.8,0-16,7.2-16,16V240h-96.5c-8.8,0-16,7.2-16,16c0,4.4,1.8,8.4,4.7,11.3
		c2.9,2.9,6.9,4.7,11.3,4.7H240v96.5c0,4.4,1.8,8.4,4.7,11.3c2.9,2.9,6.9,4.7,11.3,4.7c8.8,0,16-7.2,16-16V272h96.5
		c8.8,0,16-7.2,16-16C384.5,247.2,377.3,240,368.5,240z\\"></path>
  </g>
</svg>"
`;

exports[`ionicons5 works 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" viewBox=\\"0 0 512 512\\">
  <path d=\\"M256 112a56 56 0 1 1 56-56a56.06 56.06 0 0 1-56 56z\\" fill=\\"currentColor\\"></path>
  <path d=\\"M432 112.8l-.45.12l-.42.13c-1 .28-2 .58-3 .89c-18.61 5.46-108.93 30.92-172.56 30.92c-59.13 0-141.28-22-167.56-29.47a73.79 73.79 0 0 0-8-2.58c-19-5-32 14.3-32 31.94c0 17.47 15.7 25.79 31.55 31.76v.28l95.22 29.74c9.73 3.73 12.33 7.54 13.6 10.84c4.13 10.59.83 31.56-.34 38.88l-5.8 45l-32.19 176.19q-.15.72-.27 1.47l-.23 1.27c-2.32 16.15 9.54 31.82 32 31.82c19.6 0 28.25-13.53 32-31.94s28-157.57 42-157.57s42.84 157.57 42.84 157.57c3.75 18.41 12.4 31.94 32 31.94c22.52 0 34.38-15.74 32-31.94a57.17 57.17 0 0 0-.76-4.06L329 301.27l-5.79-45c-4.19-26.21-.82-34.87.32-36.9a1.09 1.09 0 0 0 .08-.15c1.08-2 6-6.48 17.48-10.79l89.28-31.21a16.9 16.9 0 0 0 1.62-.52c16-6 32-14.3 32-31.93S451 107.81 432 112.8z\\" fill=\\"currentColor\\"></path>
</svg>"
`;

exports[`material works 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" viewBox=\\"0 0 24 24\\">
  <path d=\\"M22 11h-4.17l3.24-3.24l-1.41-1.42L15 11h-2V9l4.66-4.66l-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93L6.34 4.34L11 9v2H9L4.34 6.34L2.93 7.76L6.17 11H2v2h4.17l-3.24 3.24l1.41 1.42L9 13h2v2l-4.66 4.66l1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24l1.42-1.41L13 15v-2h2l4.66 4.66l1.41-1.42L17.83 13H22z\\" fill=\\"currentColor\\"></path>
</svg>"
`;

exports[`tabler works 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" viewBox=\\"0 0 24 24\\">
  <g fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\">
    <path d=\\"M3 16v-5.5a2.5 2.5 0 0 1 5 0V16m0-4H3\\"></path>
    <path d=\\"M12 6v12\\"></path>
    <path d=\\"M16 16V8h3a2 2 0 0 1 0 4h-3m3 0a2 2 0 0 1 0 4h-3\\"></path>
  </g>
</svg>"
`;
